hera_cal/vis_clean.py

```diff
@@ -662,10 +752,143 @@ def fourier_filter(self, filter_centers, filter_half_widths, mode,
                         w, _ = zeropad_array(w, zeropad=zeropad[m], axis=m)
                         xp[m] = np.hstack([x[m].min() - (np.arange(zeropad[m])[::-1] + 1) * np.mean(np.diff(x[m])),
                                            x[m], x[m].max() + (1 + np.arange(zeropad[m])) * np.mean(np.diff(x[m]))])
-            mdl, res, info = dspec.fourier_filter(x=xp, data=d, wgts=w, filter_centers=filter_centers,
+            mdl, res = np.zeros_like(d), np.zeros_like(d)
```

Comment:
Lines 755 -- 884 is a large chunk of code that is only used for the specific case of this edge flagging, and is not relevant to other use-cases in Fourier filter. It also makes this "wrapper" method more of a behemoth and thus harder to maintain. I would ask that you make two separate, module-level functions either here in vis_clean or in utils, e.g.
```
def flag_edge_channels(data_in):
    """this is for skip_flagged_edge_freqs and skip_flagged_edge_times"""

def flag_edge_distance(data_in):
    """ this is for flag_within_edge_distance"""
```

and then all of this code here simply becomes
if skip_flagged_edge_freqs or skip_flagged_edge_times:
   # call flag_edge_channels

if flag_within_edge_distance:
    # call flag_edge_distance

You'll notice that for the default case of CLEAN, currently a lot of this code is actually run but none of it effects the actual data. This will make the separation between the different methods a lot cleaner (pun intended) I think.

Lmk your thoughts

Reply:
I've moved all of this code into three different breakout util functions.